### PR TITLE
Refactor converter about get_input_model_registered_name and get_output_model_registered_name_and_config

### PR DESCRIPTION
**Pull request by lvhan028**
## Motivation

turbomind.py duplicates some codes in converter.py. 
Refactor them to share the same code snippet.

### CODE REVIEW

**Review by AllentDan — APPROVED** [review]
Tested OK with:
- Qwen1.5-1.8B-Chat-AWQ
- internlm2-chat-1_8b
- Meta-Llama-3-8B-Instruct

File: lmdeploy/turbomind/deploy/converter.py [hunk 1]
```diff
@@ -124,58 +115,58 @@ def copy_tokenizer(model_path: str, tokenizer_path: str,
...
-        'w4': 'int4',
-        'w8': 'int8'
-    }
-    config.weight_type = WEIGHT_TYPE_MAP[output_format]
+            torch_dtype = getattr(model_config, 'torch_dtype', 'float16')
+            # Qwen-1 didn't set torch_dtype. It used bf16 as default
+            if model_arch == 'QWenLMHeadModel':
+                torch_dtype = 'bfloat16'
+            if not torch.cuda.is_bf16_supported():
+                print(
```
[the remark this answers is not in the document]
Comment (lvhan028) [reply lmdeploy/turbomind/deploy/converter.py:155]: OK. I'll change to logger

**Comment by lvhan028** [comment]
With the following models, I've tested:
- apply `pipeline` to transformers models directly
- convert the models and apply the pipeline
```
models=(
    "internlm2;/workspace/models-140/InternLM/internlm2-chat-7b"
    "vicuna;/workspace/models-140/vicuna/vicuna-7b-hf/"
    "base;/workspace/models-140/Yi/Yi-6B"
    "deepseek-coder;/workspace/models-140/deepseek/deepseek-coder-1.3b-instruct/"
    "deepseek;/workspace/models-140/deepseek/deepseek-llm-7b-chat/"
    "qwen;/workspace/models-140/Qwen/Qwen-7B-Chat/"
    "qwen;/workspace/models-140/Qwen/Qwen1.5-7B-Chat/"
    "llama3;/workspace/models-140/llama3/Meta-Llama-3-8B-Instruct"
    "qwen;/workspace/models-140/Qwen/Qwen1.5-4B-Chat-AWQ"
    "llama3;/workspace/models-140/llama3/Meta-Llama-3-8B-Instruct-hf-AWQ"
    "internlm2;/workspace/models-140/InternLM/internlm2-chat-1_8b-AWQ"
)
```

- apply pipeline on llava-34b and llava-34b-4bit

File: lmdeploy/turbomind/deploy/converter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lvhan028) [reply lmdeploy/turbomind/deploy/converter.py:156]: The entire converter module(lmdeploy/turbomind/deploy/*) uses print. 
If I only change the logger of converter.py, the style is inconsistent. 
If I change the logger of the entire module, it affacts `tprint`
```
def tprint(*args, **kwargs):
    to_file = kwargs.pop('to_file', False)
    if not to_file:
        return
    from io import StringIO
    s = StringIO()
    print(*args, **kwargs, file=s, end='')
    tqdm.tqdm.write(s.getvalue())
```
So I think we'd better keep `print` as we do in the `lite` module

File: lmdeploy/turbomind/turbomind.py [hunk 2]
```diff
@@ -238,53 +239,35 @@ def _from_hf(self, model_source: ModelSource, model_path: str,
...
-        self.config = cfg
-        self.data_type = data_type
-
-        logger.warning(f'model_config:\n\n{cfg.toini()}')
+        output_model_name, cfg = get_output_model_registered_name_and_config(
+            model_path=model_path,
+            model_format=engine_config.model_format,
+            group_size=0)
+        self.config = TurbomindModelConfig.from_engine_config(engine_config)
+        self.config.update(cfg)
```
Comment (irexyc) [inline lmdeploy/turbomind/turbomind.py:253]: If the user doesn't set `session_len` in backend_config, the `num_tokens_per_iter` and `max_prefill_iters` will not set in `TurbomindModelConfig.from_engine_config`.

May move [this code](https://github.com/InternLM/lmdeploy/blob/42267a8f3bf11c2579dd073e30bee95bd0149cba/lmdeploy/turbomind/deploy/target_model/base.py#L100-L106) to `update` function.

**Comment by irexyc** [comment]
https://github.com/lvhan028/lmdeploy/blob/refactor-converter/lmdeploy/utils.py#L275

If `hf_tm_config` is `TurbomindModelConfig`, should return `hf_tm_config.session_len`. It seems some model will get `max_position_embeddings: 0` like  https://github.com/InternLM/lmdeploy/issues/1703#issuecomment-2146456830

**Comment by AllentDan** [comment]
```shell
lmdeploy chat turbomind internlm-xcomposer2-vl-7b-4bit --model-format awq
```
failed.

**Comment by irexyc** [comment]
The `export_turbomind.py` file and related import should be removed.

**Comment by lvhan028** [comment]
> ```shell
> lmdeploy chat turbomind internlm-xcomposer2-vl-7b-4bit --model-format awq
> ```
> 
> failed.

I've fixed it. please double check

**Comment by lvhan028** [comment]
> https://github.com/lvhan028/lmdeploy/blob/refactor-converter/lmdeploy/utils.py#L275
> 
> If `hf_tm_config` is `TurbomindModelConfig`, should return `hf_tm_config.session_len`. It seems some model will get `max_position_embeddings: 0` like [#1703 (comment)](https://github.com/InternLM/lmdeploy/issues/1703#issuecomment-2146456830)

I've fixed it. Please check

**Comment by lvhan028** [comment]
> `rt_turbomind.py` file and related import should be removed.

Sure. I've removed it.

**Review by AllentDan — APPROVED** [review]

File: lmdeploy/turbomind/turbomind.py (hunk 2 above)
Comment (irexyc) [reply lmdeploy/turbomind/turbomind.py:253]: It seems hasn't been fixed yet.

If the user doesn't set session_len when init the pipeline, the value of `num_tokens_per_iter` and `max_prefill_iters` are not right.

File: lmdeploy/turbomind/deploy/converter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (irexyc) [reply lmdeploy/turbomind/deploy/converter.py:156]:
```
from lmdeploy import pipeline
pipe = pipeline('/nvme/shared/vicuna-7b-v1.5/', log_level='INFO')
```


```
[llama]
model_name = vicuna
model_arch = LlamaForCausalLM
...
session_len = 4104
...
num_tokens_per_iter = 0
max_prefill_iters = 1
```

num_tokens_per_iter should be 8192

File: lmdeploy/turbomind/turbomind.py [hunk 3]
```diff
@@ -20,19 +20,20 @@
...
+from .supported_models import is_supported
 from .utils import ModelSource, get_model_source
 
 # TODO: find another way import _turbomind
 lmdeploy_dir = osp.split(lmdeploy.__file__)[0]
 sys.path.append(osp.join(lmdeploy_dir, 'lib'))
 import _turbomind as _tm  # noqa: E402
 
-logger = get_logger('lmdeploy')
+logger = get_logger('lmdeploy:turbomind')
```
Comment (irexyc) [inline lmdeploy/turbomind/turbomind.py:37]: with `logger = get_logger('lmdeploy:turbomind')`, when init pipeline like `pipe = pipeline('/nvme/shared/vicuna-7b-v1.5/', log_level='INFO')`, the turbomind ini config doesn't print

Comment (lvhan028) [reply lmdeploy/turbomind/turbomind.py]: the logger is rollbacked

File: lmdeploy/turbomind/turbomind.py (hunk 2 above)
Comment (lvhan028) [reply lmdeploy/turbomind/turbomind.py:253]: Updated. It should be fixed now

**Review by irexyc — APPROVED** [review]

**Pull request merged**